experiment.py

```diff
@@ -99,17 +95,53 @@ def __init__(self, args: argparse.Namespace):
         self.neg_dist_list = []
         self.diff_dist_list = [] #pos_dist - neg_dist
         self.diff_dist_plus_margin_list = [] #pos_dist + gamma - neg_dist
+    
+    def get_dataloaders(self) -> Tuple[DataLoader, DataLoader]:
+        # TODO: pass proper args to ParaphaseDataset
+        dataset = ParaphraseDatasetElmo(
+            'quora',
+            model_name='elmo', num_examples=self.args.num_examples, 
+            max_length=self.args.max_length, stop_words_file=f'stop_words_en.txt',
+            r1=0.5, seed=self.args.random_seed
+        )
+        # Quora doesn't have a test split, so we have to do this?
+        # @js - is this right? Otherwise we should be using the actual
+        # test data from quora
+        train_dataloader, test_dataloader = train_test_split(
+            dataset, batch_size=self.args.batch_size, 
+            shuffle=True, drop_last=self.args.drop_last, 
+            train_split=self.args.train_test_split
+        )
+        return train_dataloader, test_dataloader
 
     @property
     def M(self) -> torch.nn.Parameter:
         """The orthogonal matrix, M, used for retrofitting."""
-        if self.args.model_name_or_path == 'elmo':
+        if self.args.model_name == 'elmo_single_sentence':
             return self.model.elmo._elmo_lstm._elmo_lstm.M
         else:
-            raise ValueError(f'cannot get orthogonal matrix for model {self.model_name_or_path}')
+            raise ValueError(f'cannot get orthogonal matrix for model {self.args.model_name}')
```

Comment:
is this saying that for an `elmo_sentence_pair` task you won't be able to get the `M` matrix?

Reply:
Yeah this is a good question, it's not ideal. I just have this here so we can support non-ELMO models later.  I arbitrarily decided that the model has to be called 'elmo_single_sentence' which just means ELMO when the experiment is 'retrofit'. It's not even really correct since retrofitting uses two sentences (actually four?) 

but yeah if there's a better way you can think of let me know. we could make the argparse take three models: `{'elmo_retrofit', 'elmo_single_sentence', 'elmo_sentence_pair'}` but that seems ugly too.

Reply:
Makes sense - this seems good for now.  I didn't make the connection that it defaults to `'elmo_single_sentence'` under the `retrofit` scenario.